sfdb.py

```diff
@@ -327,8 +327,11 @@ def search(self, criteria, filterFp=False):
             list: search results
         """
 
+        if not isinstance(criteria, dict):
+            return list()
+
         if list(criteria.values()).count(None) == 3:
```

Comment:
This check seems to make some assumptions about the content of the criteria dictionary (key count, valid keys, etc.). Separate arguments/keywords for the search criteria would be safer.

Reply:
This check is existing code outside of this PR. Yes, it could be approved.